nucypher/blockchain/eth/interfaces.py

```diff
@@ -280,21 +279,26 @@ def attach_middleware(self):
             self.log.debug('Injecting POA middleware at layer 0')
             self.client.inject_middleware(geth_poa_middleware, layer=0)
 
-        # Gas Price Strategy:
-        # Bundled web3 strategies are too expensive for Infura (it takes ~1 minute to get a price),
-        # so we use external gas price oracles, instead (see #2139)
-        if isinstance(self.client, InfuraClient):
+        self.client.add_middleware(middleware.time_based_cache_middleware)
+        # self.client.add_middleware(middleware.latest_block_based_cache_middleware)
```

Comment:
What's with the commented line?

Reply:
The commented line was already there before, I just moved this block. The commented line was something that seemed to cause problems in some tests, but I haven't checked it personally.

Reply:
This line causes failed tests and nonce reuse in tests - to be addressed in another PR.

Reply:
A TODO will be helpful here, perhaps indirectly linked to #2348 etc.

Reply:
✅